magento/inventory
# Fix stock indexer is executed twice after "products" import

### Description (*)

Stock indexer is executed twice after "products" import




### Manual testing scenarios (*)


- Install Magento with MSI
- Make sure "stock index" is on "update on save" mode
- Import a valid csv file with product(s)

EXPECTED RESULTS:
Stock indexer is executed only once

ACTUAL RESULTS:
Stock indexer is executed two times

**!!!IMPORTANT!!!**
To see how many times the indexer has run, one can:
- Use debugger and watch `app/code/Magento/CatalogInventory/Model/ResourceModel/Indexer/Stock/DefaultStock::_updateIndex()`
- Apply the following patch and check var/log/system.log 
```patch
Index: app/code/Magento/CatalogInventory/Model/ResourceModel/Indexer/Stock/DefaultStock.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/code/Magento/CatalogInventory/Model/ResourceModel/Indexer/Stock/DefaultStock.php	(revision 48445369fe3b8a70e3a5d620b163dfdd840501f8)
+++ app/code/Magento/CatalogInventory/Model/ResourceModel/Indexer/Stock/DefaultStock.php	(date 1577737464390)
@@ -292,6 +292,7 @@
      */
     protected function _updateIndex($entityIds)
     {
+        \Magento\Framework\App\ObjectManager::getInstance()->get(\Psr\Log\LoggerInterface::class)->critical('Indexer running', $entityIds);
         $this->deleteOldRecords($entityIds);
         $connection = $this->getConnection();
         $select = $this->_getStockStatusSelect($entityIds, true);

```



### Contribution checklist (*)
